src/client/common/errors/index.ts

```diff
@@ -0,0 +1,177 @@
+export function populateTelemetryWithErrorInfo(props: Partial<TelemetryErrorProperties>, error: Error) {
+    props.failed = true;
+    // Don't blow away what we already have.
+    props.failureCategory = props.failureCategory || getErrorCategory(error);
+    if (props.failureCategory === 'unknown' && isErrorType(error, FetchError)) {
+        props.failureCategory = 'fetcherror';
+    }
+    props.stackTrace = serializeStackTrace(error);
+    const stdErr = error instanceof BaseError ? error.stdErr : '';
+    if (!stdErr) {
+        return;
+    }
+    props.failureSubCategory = props.failureSubCategory || getReasonForKernelToDie(stdErr);
+    const info = getLastFrameFromPythonTraceback(stdErr);
+    if (!info) {
+        return;
+    }
+    props.pythonErrorFile = props.pythonErrorFile || getTelemetrySafeHashedString(info.fileName);
+    props.pythonErrorFolder = props.pythonErrorFolder || getTelemetrySafeHashedString(info.folderName);
+    props.pythonErrorPackage = props.pythonErrorPackage || getTelemetrySafeHashedString(info.packageName);
+}
+
+function parseStack(ex: Error) {
+    // Work around bug in stackTrace when ex has an array already
+    if (ex.stack && Array.isArray(ex.stack)) {
+        const concatenated = { ...ex, stack: ex.stack.join('\n') };
+        return stackTrace.parse(concatenated);
+    }
+    return stackTrace.parse(ex);
+}
+
+function serializeStackTrace(ex: Error): string {
+    // We aren't showing the error message (ex.message) since it might contain PII.
+    let trace = '';
+    for (const frame of parseStack(ex)) {
+        const filename = frame.getFileName();
+        if (filename) {
+            const lineno = frame.getLineNumber();
+            const colno = frame.getColumnNumber();
+            trace += `\n\tat ${getCallsite(frame)} ${filename}:${lineno}:${colno}`;
+        } else {
+            trace += '\n\tat <anonymous>';
+        }
+    }
+    // Ensure we always use `/` as path separators.
+    // This way stack traces (with relative paths) coming from different OS will always look the same.
+    return trace.trim().replace(/\\/g, '/');
+}
+
+function getCallsite(frame: stackTrace.StackFrame) {
+    const parts: string[] = [];
+    if (typeof frame.getTypeName() === 'string' && frame.getTypeName().length > 0) {
+        parts.push(frame.getTypeName());
+    }
+    if (typeof frame.getMethodName() === 'string' && frame.getMethodName().length > 0) {
+        parts.push(frame.getMethodName());
+    }
+    if (typeof frame.getFunctionName() === 'string' && frame.getFunctionName().length > 0) {
+        if (parts.length !== 2 || parts.join('.') !== frame.getFunctionName()) {
+            parts.push(frame.getFunctionName());
+        }
+    }
+    return parts.join('.');
+}
+
+/**
+ * Analyze the details of the error such as `stdErr` from the kernel process and
+ * try to determine the cause.
+ */
+function getReasonForKernelToDie(stdErr: string) {
```

Comment:
Final comment to make here, and this is more of a suggestion / thought to talk over. Considering if these should not be a single reason, but more like this:

pattern => tag
For all patterns that match, add all tags

Right now there is only one reason, but multiple patterns could match. Some are just searching for text that could be anywhere in the stack, while some are searching for errors at the end. If we start adding more of these ordering in this file could matter. What is something was looking for a specific string in the stacks, but that same stderr ends with a DLL load failed error. We might want to tag it with both dll load failed and the other issue. But right now it would matter which if statement runs first in this file.

Reply:
Fixed, please check

Reply:
& agree, it should be easy to add them (manage)